pandas/core/frame.py

```diff
@@ -707,11 +708,14 @@ def to_string(
         max_cols=None,
         show_dimensions=False,
         decimal=".",
+        max_colwidth=9999999,
```

Comment:
Wanted to note - one of the commenters on the issue asks: "With Pandas 0.25.0, setting display.max_colwidth to a large number stops the truncation but when trying to left justify columns with df.to_string(justify='left'), that same display setting somehow pads columns on the left so they are not left aligned. Is there any present way to prevent truncation and get left justified string columns when output to a terminal? I know a pull request is in process but I would like to do this now. Thanks."

I can accomplish this in my testing by setting ```max_colwidth=0```, which switches the padding to left.

Reply:
I'm not sure I understand this cause, but I don't think we'll want a workaround like this.

Can you post the output with `max_colwidth=0`, and how it compares to this?

Reply:
This shows how the max_colwidth=0 strangely forces a left justification:

```
>>> print(df.to_string(max_colwidth=0))
        A       B
0 NaN      NaN   
1 -1.0000  foo   
2 -2.1234  foooo 
3  3.0000  fooooo
4  4.0000  bar   
```

Whereas the behavior in this PR preserves the right justification that is the current default:

```
>>> print(df.to_string(max_colwidth=99999))
        A       B
0     NaN     NaN
1 -1.0000     foo
2 -2.1234   foooo
3  3.0000  fooooo
4  4.0000     bar
```

And interestingly, passing ```justify='left'``` doesn't have an effect:

```
>>> print(df.to_string(justify='left'))
   A      B      
0     NaN     NaN
1 -1.0000     foo
2 -2.1234   foooo
3  3.0000  fooooo
4  4.0000     bar
```

This is because of these lines in _make_fixed_width:

```
def _make_fixed_width(
    ...
    max_len = max(adj.len(x) for x in strings)
    ...
    def just(x):
        if conf_max is not None:
            if (conf_max > 3) & (adj.len(x) > max_len):
                x = x[: max_len - 3] + "..."
        return x

    strings = [just(x) for x in strings]
    result = adj.justify(strings, max_len, mode=justify)
    return result
```

It checks if ```conf_max > 3``` to apply the dot truncation ... so if it's <= 3 then that isn't called.  So it's not just 0 but any of 0, 1, 2, or 3 that causes the justification to line up.

I can spend some more time to better understand why this is happening. I agree that we should not rely on some incidentals of the underlying implementation to determine whether to justify the text.

Reply:
Ok, I figured out the issue. There are two lines where ```format_array``` is called and the ```justify``` parameter is not passed all the way through -- so in some places, the justification is being overridden.

Note that the bug where justification doesn't happen if conf_max < 3 already appears - so I think it can probably be pulled out as a separate PR.

Reply:
Scratch all that. I re-read the docs and I see that I misinterpreted the ```justify``` param, as it only refers to the column headers, not the content. In that regard it is behaving correctly. So I think I'll leave the justification question out of this PR.